### PR TITLE
charm show - tags may not exist

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/173/

**Comment by jrwren** [comment]
Some older charms do not have `Tags`. This handles that case.

Before: 
```
$ charm show cs:~evarlast/trusty/elasticsearch-3
panic: interface conversion: interface is nil, not []interface {}

goroutine 1 [running]:
panic(0x4b9de0, 0xc4201e8380)
	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/juju/charmstore-client/cmd/charm/charmcmd.newShowData(0x783440, 0xc42002c010, 0xc420220ae0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/jrwren/go/src/github.com/juju/charmstore-client/cmd/charm/charmcmd/show.go:253 +0x9b0
github.com/juju/charmstore-client/cmd/charm/charmcmd.(*showCommand).FormatSummaryTabular(0xc420262ee0, 0x783440, 0xc42002c010, 0x4b4400, 0xc420220ae0, 0xa084, 0x4a7c80)
	/Users/jrwren/go/src/github.com/juju/charmstore-client/cmd/charm/charmcmd/show.go:209 +0x8f
github.com/juju/charmstore-client/cmd/charm/charmcmd.(*showCommand).FormatSummaryTabular-fm(0x783440, 0xc42002c010, 0x4b4400, 0xc420220ae0, 0x0, 0xb00000c420060400)
	/Users/jrwren/go/src/github.com/juju/charmstore-client/cmd/charm/charmcmd/show.go:76 +0x52
github.com/juju/cmd.(*Output).writeFormatter(0xc420262ee0, 0xc4201f1770, 0xc420260910, 0x4b4400, 0xc420220ae0, 0x1, 0x0, 0x0)
	/Users/jrwren/go/src/github.com/juju/cmd/output.go:194 +0x9b
github.com/juju/cmd.(*Output).Write(0xc420262ee0, 0xc4201f1770, 0x4b4400, 0xc420220ae0, 0xc420260880, 0x0)
	/Users/jrwren/go/src/github.com/juju/cmd/output.go:169 +0xe2
github.com/juju/charmstore-client/cmd/charm/charmcmd.(*showCommand).Run(0xc420262ee0, 0xc4201f1770, 0x0, 0x0)
	/Users/jrwren/go/src/github.com/juju/charmstore-client/cmd/charm/charmcmd/show.go:159 +0x642
github.com/juju/cmd.(*SuperCommand).Run(0xc420268120, 0xc4201f1770, 0xc4201f1770, 0x0)
	/Users/jrwren/go/src/github.com/juju/cmd/supercommand.go:456 +0x319
github.com/juju/cmd.Main(0x78aee0, 0xc420268120, 0xc4201f1770, 0xc42000c130, 0x2, 0x2, 0x0)
	/Users/jrwren/go/src/github.com/juju/cmd/cmd.go:280 +0x27e
main.main()
```

After:

```
$ charm show cs:~evarlast/trusty/elasticsearch-3
Open Source, Distributed, RESTful, Search Engine built on Apache Lucene

Name			elasticsearch
Owner			evarlast
Revision		3
Supported Series	trusty
Tags
Subordinate		false
Promulgated		false
Home page
Bugs url
Read			everyone, evarlast
Write			evarlast

CHANNEL			CURRENT
stable			true
edge			true
```

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/174/

**Review by fabricematrat — APPROVED** [review]

**Review by bac — APPROVED** [review]

File: cmd/charm/charmcmd/show.go [hunk 1]
```diff
@@ -250,7 +250,9 @@ func newShowData(out io.Writer, metadata map[string]interface{}) showData {
 	if val, ok := metadata["charm-metadata"]; ok {
 		charmMetadata := val.(map[string]interface{})
 		sd.summary = charmMetadata["Summary"].(string)
-		sd.tags = toStringArray(charmMetadata["Tags"].([]interface{}))
+		if val, ok := charmMetadata["Tags"]; ok {
```
Comment (mhilton) [inline cmd/charm/charmcmd/show.go]: how about?
```
if tags, ok := charmMetadata["Tags"].([]interface{}); ok {
	sd.tags = toStringArray(tags)
}
```
That copes with a null Tags value too, although I hope you'd never receive one.

**Review by mhilton — APPROVED** [review]
LGTM with an alternative suggestion.

File: cmd/charm/charmcmd/show.go (hunk 1 above)
Comment (jrwren) [reply cmd/charm/charmcmd/show.go]: WFM

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/177/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**